jetty-maven-plugin/src/main/java/org/eclipse/jetty/maven/plugin/JettyRunMojo.java

```diff
@@ -308,15 +315,86 @@ public void configureWebApplication() throws Exception
                 }
             }
         }
+        List<Overlay> warProjectConfiguration = getWarProjectConfiguration();
+        configureWebInf(warProjectConfiguration);
+        configureBaseResource(warProjectConfiguration);
 
-        //process any overlays and the war type artifacts
-        List<Overlay> overlays = getOverlays();
-        unpackOverlays(overlays); //this sets up the base resource collection
+        // this is to handle WARs that are not referenced in maven reactor
+        List<Overlay> overlaysToUnpack = warProjectConfiguration.stream()
+            .filter(e -> !e.isMavenProject())
+            .collect(Collectors.toList());
+        unpackOverlays(overlaysToUnpack);
 
         getLog().info("web.xml file = " + webApp.getDescriptor());
         getLog().info("Webapp directory = " + webAppSourceDirectory.getCanonicalPath());
     }
 
+    private void configureWebInf(List<Overlay> warProjectConfiguration)
+        throws IOException
+    {
+        Set<File> webInfLib = new LinkedHashSet<>();
+        Set<File> webInfClasses = new LinkedHashSet<>();
+        for (Overlay overlay : warProjectConfiguration)
+        {
+            Artifact artifact = overlay.getArtifact();
+            MavenProject project = mavenProjectHelper.getMavenProject(artifact);
```

Comment:
Can't you do overlay.isMavenProject() instead?

Reply:
Yup - I totally could do it :))